README.md

```diff
@@ -53,18 +80,48 @@ compiled binaries inside the repository root in the `target/debug` (without
 | Introspection Engine | `./target/[debug\|release]/introspection-engine` |
 | Prisma Format        | `./target/[debug\|release]/prisma-fmt`           |
 
+## Prisma Schema Language
+
+The *Prisma Schema Language* is a library which defines the data structures and
+parsing rules for prisma files, including the available database connectors. For
+more technical details, please check the [library README](./psl/README.md).
+
+The PSL is used throughout the migration and introspection engines, as well as
+prisma format. The DataModeL (DML), which is an annotated version of the PSL is
+also used as input for the query engine.
+
 ## Query Engine
 
+The *Query Engine* is how Prisma Client queries are executed. Here's a brief
+description of what it does:
+- takes as inputs an annotated version of the Prisma Schema file called the
+    DataModeL (DML),
+- using the DML (specifically, the datasources and providers), it builds up a
+    [GraphQL](https://graphql.org) model for queries and responses,
+- runs as a server listening for GraphQL queries,
+- it translates the queries to the respective native datasource(s) and
+    returns GraphQL responses, and
+- handles all connections and communication with the native databases.
+
+When used through the Prisma Client, there are two ways for the Query Engine to
+be executed:
+- as a binary, downloaded during installation, launched at runtime;
+    communication happens via HTTP (`./query-engine/query-engine`)
+- as a native, platform-specific Node.js addon; also downloaded during
+    installation (`./query-engine/query-engine-node-api`)
+
 ### Usage
 
-The Query Engine can be run as a graphql server without using the Prisma Client.
-If using it on production please be aware the api and the query language can
-change any time. There is no guaranteed API stability.
+You can also run the Query Engine as a stand-alone GraphQL server.
+
+**Warning**: There is no guaranteed API stability. If using it on production
+please be aware the api and the query language can change any time.
 
 Notable environment flags:
 
 - `RUST_LOG_FORMAT=(devel|json)` sets the log format. By default outputs `json`.
-- `QE_LOG_LEVEL=(info|debug|trace)` sets the log level for the Query Engine. If you need Query Graph debugging logs, set it to "trace"
+- `QE_LOG_LEVEL=(info|debug|trace)` sets the log level for the Query Engine. If
```

Comment:
What's the default level? I'm investigating an issue related to this, I can add it myself once I know.